src/controllers/adminController.js

```diff
@@ -0,0 +1,91 @@
+const admin = require('../config/firebase');
+const db = admin.firestore();
+
+// Middleware to verify admin user
+const verifyAdmin = async (req, res, next) => {
+  const idToken = req.headers.authorization.split('Bearer ')[1];
```

Comment:
Is this going to throw an error if I don't have an authorization header? It DEFINITELY will if I have a authorization header that is an empty string (can't get `[1]` since it'll just be `[""]`).

Reply:
We should consider returning a different http code or message if the user did not provide a token for clarity of the problem. Maybe a `400 Missing token` if the token was not found in header vs `403` if the token is invalid or if we have problems verifying it.